omegaconf/resolvers/oc/__init__.py

```diff
@@ -46,8 +49,42 @@ def decode(expr: Optional[str], _parent_: Container) -> Any:
     return _get_value(val)
 
 
+def deprecated(
+    key: str,
+    message: str = "'$KEY' is deprecated. Change your code and config to use '$NEW_KEY'",
```

Comment:
Out of curiosity (no need to change it) is there a reason to prefer a string Template vs. regular string formatting, i.e.:
```python
    message: str = "'{old_key}' is deprecated. Change your code and config to use '{new_key}'",
```

By the way I have a slight preference to use `$OLD_KEY` instead of `$KEY` because here `key` is the new key (but one naturally tends to think that `key` is mapped to `$KEY` because they have the same name)

Reply:
1. no strong reason. I used string Template when formatting exception messages and this was similar.
In terms of user facing syntax, $NEW_KEY feels cleaner to me than {new_key}. it's also more explicit and does not establish a protocol via local variable names in some function. It also does not potentially leak the value of other local variables).
2. changed to OLD_KEY.